web-common/src/features/dashboards/state-managers/loaders/DashboardStateDataLoader.ts

```diff
@@ -254,14 +206,51 @@ export class DashboardStateDataLoader {
     const backButtonUsed = type === "popstate";
     const skipSessionStorage = backButtonUsed;
 
-    const exploreStateFromSessionStorage =
-      getPartialExploreStateFromSessionStorage(
-        this.exploreName,
-        this.storageNamespacePrefix,
-        urlSearchParams,
-        metricsViewSpec,
-        exploreSpec,
-      );
+    return this.getMergedExploreState({
+      metricsViewSpec,
+      exploreSpec,
+      urlSearchParams,
+      bookmarkOrTokenExploreState: this.bookmarkOrTokenExploreState
+        ? get(this.bookmarkOrTokenExploreState).data
+        : null,
+      exploreStateFromYAMLConfig,
+      rillDefaultExploreState,
+      backButtonUsed,
+      skipSessionStorage,
+    });
+  }
+
+  private getMergedExploreState({
```

Comment:
It feels a little off that this function wraps the `cascadingExploreStateMerge` function. It feels like there are two layers where there should be just one layer. Thoughts?

Reply:
They are independent. `cascadingExploreStateMerge` merges various explore states, it is based on the structure of the explore state. `getMergedExploreState` determines the order based on business logic. Would be good to keep the independent.